internal/enricher/enricher.go

```diff
@@ -0,0 +1,9 @@
+package enricher
+
+import (
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+type PipelineEnricher interface {
+	Enrich(sourceId string, request *types.GatewayBatchRequest) error
```

Comment:
please add some doc explaining what does enricher does, how it affects the request